### PR TITLE
.Net: Extract semantic functions from Kernel.Core

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

  1. Why is this change required? This is part of the work to extract functionality from Semantic Kernel core prior to 1.0 release.
  2. What problem does it solve? Isolates the semantic function implementation from the Semantic Kernel core.
  3. What scenario does it contribute to? Contributes to the goal of reducing the Semantic Kernel core to focus on AI orchestration.
  4. If it fixes an open issue, please link to the issue here. https://github.com/microsoft/semantic-kernel/issues/2563

### Description

- Move classes from abstractions SemanticFunctions folder to skills core
- Move classes from core SemanticFunctions and SkillDefinition folders to skills core
- Fix up all unit tests

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/SemanticKernel/Kernel.cs [hunk 1]
```diff
@@ -263,6 +242,7 @@ public void Dispose()
     private readonly IAIServiceProvider _aiServiceProvider;
     private readonly ILogger _logger;
 
+    /* TODO Move this
```
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel/Kernel.cs:245]: Remove comment

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/Skills/Skills.Core/Skills.Core.csproj
[the remark this answers is not in the document]
Comment (RogerBarreto) [reply dotnet/src/Skills/Skills.Core/Skills.Core.csproj:1]: Resolved with no feedback?

[the remark this answers is not in the document]
Comment (markwallace-microsoft) [reply dotnet/src/Skills/Skills.Core/Skills.Core.csproj:1]: Moving the code to `Skills.Core` was the guidance I received from @shawncal. The thinking is that we don't want to have too many small assemblies.

File: dotnet/src/Skills/Skills.Core/Skills/Core/ConversationSummarySkill.cs
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Core/Skills/Core/ConversationSummarySkill.cs:1]: `Skills/Skills.Core/Skills/Core` -- did this need to be done? Will this change with the #2733?

File: dotnet/src/Skills/Skills.Core/Skills.Core.csproj
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Core/Skills.Core.csproj:1]: Feels odd to put it in Skills.Core as these are Skills|Plugins today compared to `SemanticFunction` which is an ISKFunction implementation and some Kernel Extension methods. Maybe it's time to consider a new place where future ISKFunction implementations would go (GuidanceSemanticFunction?)

Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.Core/Skills.Core.csproj:1]: Microsoft.SemanticKernel.SkillDefinition[.Extensions].Semantic maybe?

Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.Core/Skills.Core.csproj:1]: And if keeping here, should probably update the Description [and Title].

**Review by lemillermicrosoft — APPROVED** [review]
Few comments on placement but otherwise LGTM.

File: dotnet/src/Skills/Skills.Core/Skills/Core/ConversationSummarySkill.cs
Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.Core/Skills/Core/ConversationSummarySkill.cs:1]: I'm connecting the dots, I believe this is where things are headed? I like this! re #2668

https://github.com/microsoft/semantic-kernel/blob/9529c6e87520de424047863b7b293744b5a7f4c1/docs/decisions/diagrams/skfunctions-v1.png

File: dotnet/src/Skills/Skills.Core/Skills.Core.csproj
Comment (markwallace-microsoft) [reply dotnet/src/Skills/Skills.Core/Skills.Core.csproj:1]: Will redo this PR based on this ADR https://github.com/microsoft/semantic-kernel/pull/2668

**Comment by markwallace-microsoft** [comment]
Will create a new PR once Skill -> Plugin rename work is complete. Will follow the pattern in this ADR https://github.com/microsoft/semantic-kernel/pull/2668

**Pull request closed without merging**